python/lsst/sphgeom/angle.cc

```diff
@@ -0,0 +1,101 @@
+/*
+ * LSST Data Management System
+ * See COPYRIGHT file at the top of the source tree.
+ *
+ * This product includes software developed by the
+ * LSST Project (http://www.lsst.org/).
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the LSST License Statement and
+ * the GNU General Public License along with this program.  If not,
+ * see <https://www.lsstcorp.org/LegalNotices/>.
+ */
+#include "pybind11/pybind11.h"
+
+#include "lsst/sphgeom/Angle.h"
+#include "lsst/sphgeom/NormalizedAngle.h"
+
+namespace py = pybind11;
+using namespace pybind11::literals;
+
+namespace lsst {
+namespace sphgeom {
+namespace {
+
+PYBIND11_PLUGIN(angle) {
+    py::module mod("angle");
+
+    py::class_<Angle> cls(mod, "Angle");
+
+    cls.def_static("nan", &Angle::nan);
+    cls.def_static("fromDegrees", &Angle::fromDegrees);
+    cls.def_static("fromRadians", &Angle::fromRadians);
+
+    cls.def(py::init<>());
+    cls.def(py::init<double>(), "radians"_a);
+    cls.def(py::init<Angle>(), "angle"_a);
+    // Construct an Angle from a NormalizedAngle, enabling implicit
+    // conversion from NormalizedAngle to Angle in python via
+    // py::implicitly_convertible
+    cls.def("__init__",
+            [](Angle &self, NormalizedAngle &a) {
+                new (&self) Angle(a.asRadians());
+            },
+            "normalizedAngle"_a);
```

Comment:
`NormalizedAngle` provides a cast operator for conversion to `Angle`, so this isn't needed on the C++ side. This was added on the Python side so that `py::implicitly_convertible` can work.